ifrs17/Import/ImportStorage.ipynb

```diff
@@ -329,12 +330,17 @@
                 "\n        return (ret.Any() ? ret : yc.Values.Last().RepeatOnce()).ToArray();",
                 "\n    }",
                 "\n    ",
+                "\n    // Modify this getter",
                 "\n    public YieldCurve GetYieldCurve(ImportIdentity id, string economicBasis) => (economicBasis, GetYieldCurvePeriod(id)) switch {",
                 "\n            (EconomicBases.C, PeriodType.BeginningOfPeriod ) => GetShift(id.ProjectionPeriod) > 0 ",
-                "\n                                     ? CurrentYieldCurve[id.DataNode][CurrentPeriod]",
-                "\n                                     : CurrentYieldCurve[id.DataNode][PreviousPeriod],",
-                "\n            (EconomicBases.C, PeriodType.EndOfPeriod) => CurrentYieldCurve[id.DataNode][CurrentPeriod],            ",
-                "\n            (EconomicBases.L, _ ) => LockedInYieldCurve[id.DataNode],",
+                "\n                                     ? GetCurrentYieldCurve(id.DataNode, CurrentPeriod) ",
+                "\n                                     : GetCurrentYieldCurve(id.DataNode, PreviousPeriod),",
+                "\n            (EconomicBases.C, PeriodType.EndOfPeriod) => GetCurrentYieldCurve(id.DataNode, CurrentPeriod),            ",
+                "\n            (EconomicBases.L, _ ) => LockedInYieldCurve.TryGetValue(id.DataNode, out var yc) && yc != null ? yc : ",
+                "\n                                    (YieldCurve)ApplicationMessage.Log(Error.YieldCurveNotFound, id.DataNode, DataNodeDataBySystemName[id.DataNode].ContractualCurrency, ",
+                "\n                                                                DataNodeDataBySystemName[id.DataNode].Year.ToString(), ",
+                "\n                                                                args.Year == DataNodeDataBySystemName[id.DataNode].Year ? args.Month.ToString() : MonthInAYear.ToString(), ",
```

Comment:
the month in the error message should always be args.Month.ToString()

Reply:
fixed